lib/Parser/XML.php

```diff
@@ -0,0 +1,275 @@
+
+            $this->root = new VCalendar([], false);
+            $this->pointer = &$this->input['value'][0];
+            $this->parseVcalendarComponents($this->root);
+
+        } else {
+            throw new \Exception('Unsupported XML standard');
+        }
+
+        return $this->root;
+    }
+
+    /**
+     * Parse a vCalendar.
+     *
+     * @param Sabre\VObject\Component $parentComponent
+     * @return void
+     */
+    protected function parseVcalendarComponents(Component $parentComponent) {
+
+        foreach ($this->pointer['value'] ?: [] as $children) {
+
+            switch (static::getTagName($children['name'])) {
+
+                case 'properties':
+                    $xmlProperties = $children['value'];
+
+                    foreach ($xmlProperties as $xmlProperty) {
+
+                        $propertyName       = static::getTagName($xmlProperty['name']);
+                        $propertyValue      = [];
+                        $propertyParameters = [];
+                        $propertyType       = 'text';
+
+                        foreach ($xmlProperty['value'] as $i => $xmlPropertyChild) {
+
+                            if (   !is_array($xmlPropertyChild)
+                                || 'parameters' !== static::getTagName($xmlPropertyChild['name']))
+                                continue;
+
+                            $xmlParameters = $xmlPropertyChild['value'];
+
+                            foreach ($xmlParameters as $xmlParameter) {
+                                $propertyParameters[static::getTagName($xmlParameter['name'])]
+                                    = $xmlParameter['value'][0]['value'];
+                            }
+
+                            array_splice($xmlProperty['value'], $i, 1);
+
+                        }
+
+                        switch ($propertyName) {
+
+                            case 'geo':
+                                $propertyType               = 'float';
+                                $propertyValue['latitude']  = 0;
+                                $propertyValue['longitude'] = 0;
+
+                                foreach ($xmlProperty['value'] as $xmlRequestChild) {
+                                    $propertyValue[static::getTagName($xmlRequestChild['name'])]
+                                        = $xmlRequestChild['value'];
+                                }
+                                break;
+
+                            case 'request-status':
+                                $propertyType = 'text';
+
+                                foreach ($xmlProperty['value'] as $xmlRequestChild) {
+                                    $propertyValue[static::getTagName($xmlRequestChild['name'])]
+                                        = $xmlRequestChild['value'];
+                                }
+                                break;
+
+                            case 'freebusy':
+                                $propertyType = 'freebusy';
```

Comment:
For 'cases' where there's no break, I think it's important to add an inline docblock that explains why.. or at the very least one that says 'no break intentional'.